kml/types.cpp

```diff
@@ -30,6 +30,48 @@ void SetBookmarksMinZoom(FileData & fileData, double countPerTile, int maxZoom)
   minZoomQuadtree.SetMinZoom(countPerTile, maxZoom, setMinZoom);
 }
 
+bool MultiGeometry::IsValid() const
+{
+  return !m_lines.empty();
+}
+
+void MultiGeometry::Clear() {
+  m_lines.clear();
+  m_timestamps.clear();
+}
+
+bool MultiGeometry::HasTimestamps() const
+{
+  if (m_timestamps.empty())
+    return false;
+  for (auto const & timestamp : m_timestamps)
+  {
+    if (!timestamp.empty())
+      return true;
+  }
+  return false;
+}
+
+bool MultiGeometry::HasTimestampsFor(size_t lineIndex) const
+{
+  return !(m_timestamps.empty() || lineIndex >= m_timestamps.size() || m_timestamps[lineIndex].empty());
```

Comment:
These conditions should always be true, correct?

```suggestion
  ASSERT(!m_timestamps.empty(), ());
  ASSERT(m_lines.size(), m_timestamps.size(),());
  return !m_timestamps[lineIndex].empty();
```

Reply:
Yes! Thanks!